Feature/6 lightbridge authz helm chart

Franck-Sorel opened the pull request:
Local Test ran successfully !
Closes #11 

<img width="1365" height="470" alt="Screenshot from 2025-10-14 13-01-37" src="https://github.com/user-attachments/assets/b4262ecd-d7cc-43f8-a2ad-68c521756bda" />

Franck-Sorel committed:
chore: update Chart.yaml and values.yaml for lightbridge-authz-umbrella

Franck-Sorel committed:
fix: correct ClientTrafficPolicy template and values structure

- Fixed ClientTrafficPolicy template to use proper targetRefs array structure
- Corrected enableProxyProtocol to be boolean instead of nested object
- Removed invalid 'connection' field not supported in ClientTrafficPolicy API
- Updated values.yaml to match template expectations with proper nesting
- Set dynamic namespace using .Release.Namespace with default fallback
- Simplified template by removing unnecessary common.tplvalues.render calls

The template now generates valid ClientTrafficPolicy manifests that comply with
Envoy Gateway v1.4+ API specifications, resolving previous installation failures
and template evaluation errors.

Franck-Sorel committed:
fix(helm): resolve config,probes,sub-chart issues

Franck-Sorel committed:
Refactor code structure for improved readability and maintainability

Franck-Sorel committed:
Refactor move chart dependencies at the root of charts/

Franck-Sorel committed:
Ignore parent chart

Franck-Sorel committed:
Chart.yaml for lightbridge-authz

Franck-Sorel committed:
Remove umbrella chart

Franck-Sorel committed:
fix(helm): Update umbrella chart to use dependencies releases

Franck-Sorel committed:
fix(helm): Remove helm chart from .gitignore

Franck-Sorel committed:
Update(helm): Add Chart.lock

Franck-Sorel committed:
Refactor(): Move child charts under charts/

Franck-Sorel committed:
feat: add lightbridge-authz-0.1.0

Franck-Sorel committed:
feat: add lightbridge-authz-0.1.0

Franck-Sorel committed:
Update: remove uncessary artifacts for workflow to deploy to gh-pages

Franck-Sorel committed:
fix(lightbridge-config): make client-traffic-policy template more robust

Dericko681 commented on charts/lightbridge-authz-umbrella/values.yaml:
```diff
@@ -0,0 +1,57 @@
+  secret:
+    enabled: true
+    databaseUrl: "postgres://postgres:postgres@my-postgres-postgresql.default.svc.cluster.local:5432/lightbridge_authz"
+    jwksUrl: "https://your-auth-server.com/certs"
```
if u don't have a value for this can u remove the attribute?

Franck-Sorel committed:
fix(): ignore artifacts packages under .cr-release-packages/

Dericko681 reviewed (changes requested):
nice proposal

study [bjw-s](https://github.com/bjw-s-labs/helm-charts) and consider a refactor?

Franck-Sorel committed:
Merge branch 'main' into feature/6-lightbridge-authz-helm-chart

Franck-Sorel committed:
feat(authz): Improve secret management

Makes secret creation optional and allows the chart to use an existing, externally managed secret.

- The  value now controls secret creation (default: false).
- The  value specifies the name of the secret to use.
- The deployment is updated to use the specified secret name.

Franck-Sorel committed:
Merge branch 'feature/6-lightbridge-authz-helm-chart' of github.com:ADORSYS-GIS/ai-helm into feature/6-lightbridge-authz-helm-chart

Franck-Sorel committed:
feat(umbrella): Update authz dependency and configuration

- Updates the lightbridge-authz chart to a new version that uses external secrets.
- Updates the container image to ghcr.io/franck-sorel/lightbridge-authz.
- Configures the umbrella chart to use an externally managed secret for the authz service.

Franck-Sorel committed:
refactor(authz): Migrate to app-template chart

Refactors the lightbridge-authz chart to use the bjw-s/app-template for resource generation.

- Removes manual templates for Deployment, Service, ServiceAccount, Ingress, and HPA.
- Adds a new  to call the app-template.
- Restructures the  to be compatible with the app-template, including new , , , and
 sections.
- Cleans up the  file by removing old, unused values.

stephane-segning commented on charts/lightbridge-authz-umbrella/values.yaml:
```diff
@@ -0,0 +1,56 @@
+# Global values
+global:
+  namespace: lightbridge
```
Why not use the value that comes naturally with helm commands?

stephane-segning commented on charts/lightbridge-authz/templates/_helpers.tpl:
Why do you need this file?

stephane-segning replied:
Why don't you use common you added earler?

stephane-segning commented on charts/lightbridge-authz/templates/app.yaml:
Not required when using app-template from bjw

Franck-Sorel committed:
feat: Address PR feedback and align with best practices

This commit addresses feedback from the recent pull request review and improves the overall quality and consistency of the Helm charts.

The following changes have been made:

- **:**
  - Removed  and  from  to improve chart reusability and rely on Helm's default naming conventions.

- **:**
  - Removed the redundant  and  files. The functionality is now handled by the  chart, reducing boilerplate code.

- **:**
  - Aligned the model backends in  with the defined providers, correcting a routing issue where Gemini models were
  pointing to a non-existent backend.
  - Fixed a critical bug in  where the template was iterating over the wrong map, resulting in empty
   and disabling TLS validation. The template now correctly populates the , ensuring that TLS validation is
  applied as intended.

Franck-Sorel committed:
feat(lightbridge-config): Refactor and improve chart

This commit introduces a number of improvements to the `lightbridge-config` chart, making it more robust, maintainable, and aligned with best practices.

The following changes have been made:

- **Refactored Rate Limiting:**
  - Combined `rl-requests.yaml` and `rl-tokens.yaml` into a single, generic `ratelimit.yaml` template. This reduces code duplication and makes the rate limiting configuration more modular.
  - Updated `values.yaml` with a new `rateLimitPolicies` structure to support the new generic template.

- **Improved `backends.yaml`:**
  - Fixed a bug where an `AIServiceBackend` could be created without a corresponding `Backend`, leading to a dangling reference.
  - Consolidated redundant `if .endpoints` conditions to make the template cleaner and more efficient.

- **Simplified `guard-securitypolicy.yaml`:**
  - Refactored the template to improve readability and remove redundant header checks, making it more efficient and easier to understand.

Franck-Sorel committed:
feat(lightbridge-authz): Refactor to use app-template and fix tests

This commit refactors the `lightbridge-authz` chart to leverage the `bjw-s/app-template` chart, significantly reducing boilerplate
and improving maintainability.

The following changes have been made:

- **Chart Refactoring:**
  - Removed redundant `_helpers.tpl`, `configmap.yaml`, and `secret.yaml` files, as their functionality is now provided by `app-template`.
  - Simplified `values.yaml` to align with `app-template`'s configuration structure.

- **Test Fixes:**
  - Updated `test-connection.yaml` to use the correct helpers (`common.names.fullname`, `common.labels.standard`) from the `bjw-s/common` library, resolving template rendering errors.
  - Corrected YAML indentation in `test-connection.yaml` to ensure proper parsing.

Franck-Sorel committed:
feat(lightbridge-authz): adopt app-template with secure config generation

This commit refactors the `lightbridge-authz` Helm chart to fully leverage
the `bjw-s/app-template` library chart, resulting in a cleaner and more
maintainable structure. It also introduces a robust and secure method for
managing application configuration, addressing production-grade secret
handling.

- Reverted to using `app-template` as the core rendering engine for Kubernetes resources.
- Introduced `templates/main.yaml` to include `bjw-s.common.lib.all`, enabling full `app-template` functionality.
- Restored `app-template` as a dependency in `Chart.yaml`.
- Refactored `values.yaml` to conform precisely to the `app-template` schema, ensuring correct configuration of controllers, services, and persistence.

- Implemented an `initContainer` named `config-generator` to dynamically generate the `config.yaml` file.
- Securely merges sensitive data from a Kubernetes `Secret` (`lightbridge-authz-prod-secrets`) with non-sensitive configuration from a `ConfigMap` template.
- The generated `config.yaml` is written to a shared `emptyDir` volume mounted by the main application container.
- Sets the `CONFIG_PATH` environment variable to reference the dynamically created configuration file.

- Corrected the directory structure of the `app-template` subchart to ensure Helm correctly unpacks and utilizes it.

Franck-Sorel committed:
feat(lightbridge-config): add targetRef to ClientTrafficPolicy

This commit adds a `targetRef` to the `clientTraffic` section in the
`values.yaml` file. This ensures that the `ClientTrafficPolicy` is applied
to a specific gateway, allowing client traffic settings—such as
`enableProxyProtocol` and `tcpKeepalive`—to be correctly enforced.

Without this change, the `ClientTrafficPolicy` would be created but not
associated with any gateway, rendering it ineffective.

Franck-Sorel committed:
feat(umbrella): align with refactored subcharts and local dependencies

This commit updates the `lightbridge-authz-umbrella` chart to correctly use
the refactored `lightbridge-authz` and `lightbridge-config` subcharts as
local dependencies.

Key changes include:

- **Local Dependency Resolution:**
  - Updated `Chart.yaml` and `Chart.lock` to reference subcharts via local
    file paths (`file://../<chart-name>`), ensuring the umbrella chart uses
    the latest local versions instead of remote ones.

- **Values Overrides for `lightbridge-authz`:**
  - Reorganized `values.yaml` overrides to align with the new
    `app-template`-based structure.
  - Added `enabled: true` and `app-template.enabled: true` to ensure proper
    rendering of both the subchart and its dependency.

- **Values Cleanup for `lightbridge-config`:**
  - Removed obsolete and unused overrides, resulting in a cleaner and more
    maintainable `values.yaml`.

These updates ensure consistent deployment across the umbrella chart and its
refactored subcharts.

Dericko681 reviewed (changes requested):
Did the chart work for you?

<img width="841" height="160" alt="Image" src="https://github.com/user-attachments/assets/dc14bb96-616b-4696-9bbc-8c28857737e4" />


<img width="833" height="76" alt="Image" src="https://github.com/user-attachments/assets/a53a6488-0fa2-4377-870d-10858f19e0f2" />

Dericko681 commented on charts/lightbridge-authz/Chart.yaml:
```diff
@@ -0,0 +1,34 @@
+  - name: common
+    version: "*"
+    repository: https://charts.bitnami.com/bitnami
+  - name: app-template
```
why using bitnami? besides i get this

<img width="624" height="277" alt="Image" src="https://github.com/user-attachments/assets/f70476b8-e614-45d6-82c9-5aecc043d427" />

Franck-Sorel replied:
Kidding ! This  link is like the one provided by `Github pages` after a chart has been release, Not a link to access from the browser...Kindly take look, once more, at  the helm chart release process... step by step...

Dericko681 replied:
besides, this is what i am saying

<img width="842" height="284" alt="image" src="https://github.com/user-attachments/assets/7ad6744b-7465-41b1-8a23-4ad51172ef97" />



you are not using the bjws correctly else you will not need the bitnami depency. remove one dependency

Franck-Sorel replied:
The issue is not the link access on the browser, but the usage of the Bitnami common dependency which is only minimally used and can be bypass ! :+1:

Dericko681 commented on charts/lightbridge-authz/Chart.yaml:
```diff
@@ -0,0 +1,34 @@
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "1.16.0"
+
```
all these comments?

Dericko681 commented on charts/lightbridge-authz-umbrella/Chart.yaml:
```diff
@@ -0,0 +1,15 @@
+  - name: lightbridge-authz
+    version: 0.1.0
+    repository: "https://adorsys-gis.github.io/ai-helm/"
+  - name: lightbridge-config
```
you have nott modified the workflow so is lightbridge-auths available in this repo?

Franck-Sorel replied:
Are you talking about the github workflow that has to release the helm chart ? If so, kindly take a look at the [workfow](https://github.com/ADORSYS-GIS/ai-helm/blob/main/.github/workflows/release-helm-charts.yml).. The workflow will release it automatically, not need of modification ..

Dericko681 commented on charts/lightbridge-authz/values.yaml:
```diff
@@ -0,0 +1,91 @@
+app-template:
+  controllers:
```
really?

Franck-Sorel replied:
Clarify the comment !

Franck-Sorel commented on charts/lightbridge-config/Chart.yaml:
```diff
@@ -0,0 +1,29 @@
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+
+# This is the chart version. This version number should be incremented each time you make changes
```
Kind take a look at how to build a helm chart from the [official docs](https://helm.sh/docs/helm/helm_create/), comments are generated by the command `helm create`

Franck-Sorel commented on charts/lightbridge-authz/values.yaml:
```diff
@@ -0,0 +1,91 @@
+      globalMounts:
+        - path: /secrets
+          readOnly: true
+    config-template:
```
What do you means ? `/secrets` is the path ! Need more clarification for comment...

Franck-Sorel commented on charts/lightbridge-authz/Chart.yaml:
```diff
@@ -0,0 +1,34 @@
+name: lightbridge-authz
+description: A Helm chart for Kubernetes
+
+# A chart can be either an 'application' or a 'library' chart.
```
The chart need to be connected to a postgres DB and OAuth2 server, all those info has to written according to this [schema](https://github.com/ADORSYS-GIS/lightbridge-authz/blob/main/config/default.yaml) and input it as a Kubernete Secret !

Franck-Sorel committed:
docs(helm): improve chart descriptions for authz, config, and umbrella

This commit updates the descriptions in the `Chart.yaml` files for the
`lightbridge-authz-umbrella`, `lightbridge-authz`, and `lightbridge-config`
charts.

The new descriptions provide a more detailed explanation of the purpose and
function of each chart, improving the overall clarity and maintainability of
the Helm release.

- **lightbridge-authz-umbrella**: Now clearly described as the main entry point
  for deploying the entire Lightbridge authorization system.
- **lightbridge-authz**: Description updated to highlight its role as the core
  authorization service.
- **lightbridge-config**: Description updated to explain its responsibility for
  managing authorization policies, networking, and security configurations.

benie-joy-possi commented:
Don't forget to work on the linting errors we get when running the lint helm chart workflow

Franck-Sorel commented on charts/lightbridge-authz/values.yaml:
```diff
@@ -0,0 +1,91 @@
+              # Read the sensitive values from the secret
+              DATABASE_URL=$(cat /secrets/DATABASE_URL)
+              JWKS_URL=$(cat /secrets/JWKS_URL)
+  
```
He who want to install this chart has to create a Kubernetes secrets with DATABASE_URL and JWK_URL and the init container will create the associated configmap that will be use to pass to the rust application, that'show we pass the secret to the application..
An exmaple of the secrets require during the installation process :
```sh
   1 apiVersion: v1
   2 kind: Secret
   3 metadata:
   4   name: lightbridge-authz-secret
   5 type: Opaque
   6 stringData:
   7   DATABASE_URL: "postgresql://user:password@host:port/database"
   8   JWK_URL: "https://your-jwk-url.com/.well-known/jwks.json"
```

Koufan-De-King reviewed (changes requested):
I would like this to be better documented, especially how to test and get the same results–including reasons for the method of testing being valid and not casuing issues for actual prod deployment.
Also, explaining how the secrets are getting injected into our k8s cluster and then our self-service application would be great too.

stephane-segning reviewed (commented):
This calls for a demo

Franck-Sorel committed:
feat(lightbridge): add README documentation for lightbridge-authz and lightbridge-config charts

Franck-Sorel committed:
git commit -m "refactor: remove Bitnami common dependency and use standard Helm functions

- Remove Bitnami common chart dependency from Chart.yaml
- Refactor test-connection.yaml to use native Helm template functions
- Update external config documentation files"

Franck-Sorel committed:
Docs: remove comments from Chart.yaml

Dericko681 reviewed (changes requested):
good job!

 yaml [files](charts/lightbridge-authz/docs/external-config/) in docs/ ?

Franck-Sorel committed:
Remove deprecated security and rate limit configurations from lightbridge-config chart

- Deleted guard-securitypolicy.yaml, ratelimit.yaml, retry.yaml, rl-local-burst.yaml, route.yaml, and test.yaml templates.
- Removed associated values from values.yaml, including rate limit policies and model configurations.
- Cleaned up unused backend traffic policies and security policies to streamline the chart.

Franck-Sorel committed:
chore: remove Keycloak configuration files from lightbridge-authz chart

Dericko681 reviewed (approved):
good job

benie-joy-possi reviewed (commented):
Great work overall!
Just want to find out if there will be any other demo session for this! If yes i would gladly like to attend too!
